src/azure-cli-core/azure/cli/core/_identity.py

```diff
@@ -256,16 +258,15 @@ def _decode_managed_identity_token(self, credential, resource):
         return decoded
 
     def get_user(self, user_or_sp=None):
-        try:
-            return self._msal_app.get_accounts(user_or_sp)
-        except ValueError:
```

Comment:
`ValueError` is thrown due to the incorrect `authority` given to `PublicClientApplication`. If the `username` is not found, `[]` will be return.